conda_smithy/lint_recipe.py

```diff
@@ -144,994 +202,185 @@ def find_local_config_file(recipe_dir, filename):
+    """
 
-        # Combine commenters from both issue comments and review comments
-        commenters = {comment.user.login for comment in issue_comments}
-        commenters.update({review.user.login for review in review_comments})
+    warnings.warn(
+        "lintify_meta_yaml is deprecated and will be removed in v4, use lint_meta_yaml (signature changed) instead. ",
+        DeprecationWarning,
+    )
 
-        # Check if all maintainers have either commented or are the PR author
-        non_participating_maintainers = set()
-        for maintainer in maintainers:
-            if maintainer not in commenters and maintainer != pr_author:
-                non_participating_maintainers.add(maintainer)
+    recipe_dir_path = Path(recipe_dir) if recipe_dir else None
 
-        # Add a lint message if there are any non-participating maintainers
-        if non_participating_maintainers:
-            lints.append(
-                f"The following maintainers have not yet confirmed that they are willing to be listed here: "
-                f"{', '.join(non_participating_maintainers)}. Please ask them to comment on this PR if they are."
-            )
+    lint_result = lint_meta_yaml(meta, recipe_dir_path, conda_forge)
 
+    return lint_result.lints, lint_result.hints
 
-def is_selector_line(line, allow_platforms=False, allow_keys=set()):
-    # Using the same pattern defined in conda-build (metadata.py),
-    # we identify selectors.
-    line = line.rstrip()
-    if line.lstrip().startswith("#"):
-        # Don't bother with comment only lines
-        return False
-    m = sel_pat.match(line)
-    if m:
-        nouns = {
-            w for w in m.group(3).split() if w not in ("not", "and", "or")
-        }
-        allowed_nouns = (
-            {"win", "linux", "osx", "unix"} if allow_platforms else set()
-        ) | allow_keys
 
-        if nouns.issubset(allowed_nouns):
-            # the selector only contains (a boolean chain of) platform selectors
-            # and/or keys from the conda_build_config.yaml
-            return False
-        else:
-            return True
-    return False
+def is_selector_line(
+    line: str,
+    allow_platforms: bool = False,
+    allow_keys: Optional[AbstractSet] = None,
+):
+    warnings.warn(
+        "is_selector_line is deprecated and will be removed in v4, "
+        "use linters_meta_yaml.is_selector_line instead.",
+        DeprecationWarning,
+    )
+    return linters_meta_yaml.is_selector_line(
+        line, allow_platforms, allow_keys
+    )
 
 
-def is_jinja_line(line):
-    line = line.rstrip()
-    m = jinja_pat.match(line)
-    if m:
-        return True
-    return False
+def is_jinja_line(line: str) -> bool:
+    warnings.warn(
+        "is_jinja_line is deprecated and will be removed in v4. "
+        "Use linters_meta_yaml.is_jinja_line instead.",
+        DeprecationWarning,
+    )
+    return linters_meta_yaml.is_jinja_line(line)
 
 
 def selector_lines(lines):
```

Comment:
Are any of these util functions actually used outside of `conda-smithy` @beckermr? I couldn't find any uses in the conda-forge org outside of this repo. In general, I think we should design a clearly specified public API in the long-term instead of assuming that every public function in this module is a potentially public API. As far as I can tell, other components only use `conda_smithy.lint_recipe.main`.

Reply:
+1

There are good Python libraries I've seen that prefix private modules and functions with `_` to create a clear distinction between public and private API.